pkgs/swift2objc/lib/src/config.dart

```diff
@@ -0,0 +1,22 @@
+import 'dart:io';
+
+final defaultTempDir = Directory.systemTemp.uri;
+const defaultTempDirPrefix = 'swift2objc_temp_';
+const symbolgraphFileSuffix = '.symbols.json';
+const defaultModuleName = 'symbolgraph_module';
+
+class Config {
+  final List<Uri> inputFiles;
+  final Uri outputFile;
+  final Uri? tempDir;
+  final String moduleName;
+  final bool deleteTempAfterDone;
```

Comment:
I don't think you need this option. I figured you might need it for the integration test, but after rereading it that's not really true. It's a bit confusing because there's 2 concepts of "temp dir" that are being conflated. First, there's the `config.tempDir` field above. Second is the `tempDir` variable in `generateWrapper`, which is actually a subdir of `config.tempDir`. Users won't know that directory's name because it's randomly assigned, so it's hard to access programmatically (eg in an integration test).

Anyway, this option controls whether the `generateWrapper`'s `tempDir` is deleted, not whether `config.tempDir` is deleted. That's pretty confusing. I think `generateWrapper`'s `tempDir` should always be deleted, and `config.tempDir` should never be deleted. So you can just remove `deleteTempAfterDone`.

Another approach would be to set `generateWrapper`'s `tempDir` equal to `config.tempDir` if it's non-null, rather than creating a temp dir inside `config.tempDir`, and not delete this dir. If `config.tempDir` is null you'd still set `generateWrapper`'s `tempDir` to `defaultTempDir.createTemp(...)`, and delete it afterwards. Again you wouldn't need `deleteTempAfterDone`. This approach might be useful for debugging, so we can inspect the symbol graphs.

Reply:
Sound reasonable. Delete temp directory when it's created by us, leave it when it's specified by the user.